File: JavaAndScala/src/main/scala/org/lang/scala/common/utils/CollectionUtils.scala
package org.lang.scala.common.utils

/**
 * 	This is a stand-alone object that contains collection utility methods
 * 
 * 	@author VinceYuan
 */
object CollectionUtils {
  
  /**
   * 	This is a method to concatenate the items in a list
   */
  def toString(list: List[Any]): String = {
    val stringBuilder: StringBuilder = new StringBuilder()
    for (i <- 0 until list.size) {
      if (i > 0) stringBuilder.append(", ")
      stringBuilder.append(list(i))
    }
    stringBuilder.toString()
  }
}

File: JavaAndScala/src/main/java/com/java/se/common/utils/CollectionUtils.java
package com.java.se.common.utils;

import java.util.Collection;

/**
 * 	This is a class to implement Collection utility methods
 * 
 * @author VinceYuan
 *
 */
public class CollectionUtils {

	/**
	 * 	This is a method to concatenate the elements to a string
	 * @param collection
	 * @return
	 */
	public final static <T extends Collection<?>> String toString(T collection) {
		
		/*	Initialize a string buffer	*/
		StringBuffer stringBuffer = new StringBuffer();
		
		/*	Concatenate the elements to a string	*/
		int index = 0;
		for (Object element : collection) {
			if (index > 0) stringBuffer.append(", ");
			stringBuffer.append(element);
			index++;
		}
		
		/*	Return the string	*/
		return stringBuffer.toString();
	}
}
